model/proto/jaeger.proto

```diff
@@ -0,0 +1,212 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax="proto3";
+package jaeger;
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/duration.proto";
+import "google/api/annotations.proto";
+import "protoc-gen-swagger/options/annotations.proto";
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+// import "github.com/mwitkow/go-proto-validators/validator.proto";
+
+// Specifies path for generated file (relative to $GOPATH) and the name of the package.
+option go_package = "github.com/jaegertracing/jaeger/model;model";
+
+option (grpc.gateway.protoc_gen_swagger.options.openapiv2_swagger) = {
+  info: {
+    version: "1.0";
+  };
+  external_docs: {
+    url: "https://github.com/jaegertracing/jaeger";
+    description: "Jaeger API";
+  }
+  schemes: HTTP;
+  schemes: HTTPS;
+};
+
+// Enable gogoprotobuf extensions (https://github.com/gogo/protobuf/blob/master/extensions.md).
+// Enable custom Marshal method.
+option (gogoproto.marshaler_all) = true;
+// Enable custom Unmarshal method.
+option (gogoproto.unmarshaler_all) = true;
+// Enable custom Size method (Required by Marshal and Unmarshal).
+option (gogoproto.sizer_all) = true;
+// Enable registration with golang/protobuf for the grpc-gateway.
+option (gogoproto.goproto_registration) = true;
+
+message TraceID {
+  // We have a custom String() method that renders trace ID as a single hex string, so turn off auto-gen.
+  option (gogoproto.goproto_stringer) =  false;
+
+  uint64 low = 1;
+  uint64 high = 2;
+}
+
+// For comparison of how KeyValue looks when using `oneof`.
+// Both definitions result in the same message size.
+// message KeyValue2 {
+//   string key = 1;
+//   oneof value {
+//     string vStr = 2;
+//     double  vDouble = 3;
+//     bool vBool = 4;
+//     int64 vLong = 5;
+//     bytes vBinary = 6;
+//   }
+// }
+
+enum ValueType {
+  STRING  = 0;
+  BOOL    = 1;
+  INT64   = 2;
+  FLOAT64 = 3;
+  BINARY  = 4;
+};
+
+message KeyValue {
+  option (gogoproto.equal) = true;
+  option (gogoproto.compare) = true;
+
+  string    key      = 1;
+  ValueType vType    = 2;
+  string    vStr     = 3;
+  bool      vBool    = 4;
+  int64     vInt64   = 5;
+  double    vFloat64 = 6;
+  bytes     vBinary  = 7;
+}
+
+message Log {
+  google.protobuf.Timestamp timestamp = 1 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = false
+  ];
+  repeated KeyValue fields = 2 [
+    (gogoproto.nullable) = false
+  ];
+}
+
+enum SpanRefType {
+  CHILD_OF = 0;
+  FOLLOWS_FROM = 1;
+};
+
+message SpanRef {
+  TraceID traceID = 1 [
+    (gogoproto.nullable) = false
+  ];
+  uint64 spanID = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.customtype) = "SpanID" // alias to uint64
+  ];
+  SpanRefType refType = 3;
+}
+
+message Process {
+  string service_name = 1;
+  repeated KeyValue tags = 2 [
+    (gogoproto.nullable) = false
+  ];
+}
+
+message Span {
```

Comment:
Here it's all mixy with the naming :) trace_id, span_id, start_time and process_id. operation_name is the only outlier here, but the correct one.

Reply:
fixed